readthedocs/rtd_tests/tests/test_doc_building.py

```diff
@@ -147,6 +247,17 @@ def test_failing_execution(self):
 
         # api() is not called anymore, we use api_v2 instead
         self.assertFalse(self.mocks.api()(DUMMY_BUILD_ID).put.called)
+        # The command was saved
+        command = build_env.commands[0]
+        self.mocks.mocks['api_v2.command'].post.assert_called_once_with({
```

Comment:
Please, check all of these lines because maybe you are missing the `()` in them.

Reply:
I think this was because the put method was getting called with an id on `()` before sending the data, the post method just send the data.

Reply:
This is from pdb (post and put methods)

```
248             # api() is not called anymore, we use api_v2 instead
249             self.assertFalse(self.mocks.api()(DUMMY_BUILD_ID).put.called)
250             # The command was saved
251             command = build_env.commands[0]
252             __import__('pdb').set_trace()
253  ->         self.mocks.mocks['api_v2.command'].post.assert_called_once_with({
254                 'build': DUMMY_BUILD_ID,
255                 'command': command.get_command(),
256                 'description': command.description,
257                 'output': command.output,
258                 'exit_code': 1,
(Pdb++) mock = self.mocks.mocks['api_v2.command']
(Pdb++) dir(mock)
['assert_any_call', 'assert_called', 'assert_called_once', 'assert_called_once_with', 'asser
t_called_with', 'assert_has_calls', 'assert_not_called', 'attach_mock', 'call_args', 'call_a
rgs_list', 'call_count', 'called', 'configure_mock', 'get', 'method_calls', 'mock_add_spec',
 'mock_calls', 'post', 'reset_mock', 'return_value', 'side_effect']
(Pdb++) mock.post.call_args_list
[call({'start_time': datetime.datetime(2018, 3, 24, 0, 3, 39, 202898), 'command': u'echo tes
t', 'output': u'This is not okay', 'exit_code': 1, 'description': '', 'end_time': datetime.d
atetime(2018, 3, 24, 0, 3, 39, 203003), 'build': 123})]
(Pdb++) mock().post.call_args_list
[]
(Pdb++) mock = self.mocks.mocks['api_v2.build']
(Pdb++) mock.put.call_args_list
[]
(Pdb++) mock().put.call_args_list
[call({'length': 0, 'state': u'finished', 'output': u'', 'exit_code': 1, 'error': u'', 'vers
ion': 20, 'setup_error': u'', 'builder': u'stsewd', u'id': 123, 'project': 6, 'setup': u'',
'success': False})]

```